lib/explorer/polars_backend/shared.ex

```diff
@@ -135,6 +135,14 @@ defmodule Explorer.PolarsBackend.Shared do
       end
 
     Native.s_from_list_of_series_as_structs(name, series)
+    |> then(fn polars_series ->
+      if Native.s_dtype(polars_series) != dtype do
+        {:ok, casted} = Native.s_cast(polars_series, dtype)
+        casted
+      else
+        polars_series
+      end
+    end)
```

Comment:
Just to give a context, the problem was because the "empty" inner struct series were getting a different dtype, because it couldn't be "inferred" from the series alone (since it was empty).
The cast will force those empty struct series to keep the same dtype.